server/src/main/java/com/linecorp/centraldogma/server/internal/httpapi/ProjectServiceV1.java

```diff
@@ -0,0 +1,160 @@
+                                                                   .map(DtoConverter::convert)
+                                                                   .collect(Collectors.toList()));
+    }
+
+    /**
+     * POST /projects
+     *
+     * <p>Creates a new project.
+     */
+    @Post("/projects")
+    public CompletionStage<ProjectDto> createProject(AggregatedHttpMessage message) {
+        final JsonNode jsonNode = getJsonNode(message);
+        checkArgument(jsonNode.get("name") != null, "project name should be non-null");
+        final String name = jsonNode.get("name").textValue();
+        checkArgument(!isNullOrEmpty(name), "project name should be non-null");
+
+        return execute(Command.createProject(AuthenticationUtil.currentAuthor(), name))
+                .handle((unused, thrown) -> {
+                    if (thrown != null) {
+                        if (Throwables.getRootCause(thrown) instanceof StorageExistsException) {
+                            throw newHttpResponseException(HttpStatus.BAD_REQUEST,
+                                                           "project " + name + " already exists");
+                        }
+                        return Exceptions.throwUnsafely(thrown);
+                    }
+                    return DtoConverter.convert(projectManager().get(name));
+                });
+    }
+
+    /**
+     * DELETE /projects/{projectName}
+     *
+     * <p>Removes a project.
+     */
+    @Delete("/projects/{projectName}")
+    public CompletableFuture<Void> removeProject(@Param("projectName") String name) {
+        return execute(Command.removeProject(AuthenticationUtil.currentAuthor(), name))
+                .handle(voidFunction((unused, thrown) -> {
+                    if (thrown != null) {
+                        if (Throwables.getRootCause(thrown) instanceof StorageNotFoundException) {
+                            throw newHttpResponseException(HttpStatus.NOT_FOUND,
+                                                           "project " + name + " not found");
+                        }
+                        Exceptions.throwUnsafely(thrown);
+                    }
+                }));
+    }
+
+    /**
+     * PATCH /projects/{projectName}
+     *
+     * <p>Patches a project with the JSON_PATCH. Currently, only unremove project operation is supported.
+     */
+    @ConsumeType("application/json-patch+json")
+    @Patch("/projects/{projectName}")
+    public CompletionStage<ProjectDto> patchProject(
+            @Param("projectName") String name, AggregatedHttpMessage message) {
+        final JsonNode jsonNode = getJsonNode(message);
+
+        if (!jsonNode.equals(unremovePatch)) {
+            throw newHttpResponseException(HttpStatus.BAD_REQUEST,
+                                           "not supported JSON patch: " + message.content() +
+                                           " (expected: " + unremovePatch.toString() + ')');
```

Comment:
Maybe `CONFLICT`?

Reply:
Is this `CONFLICT`? This `patch` method only supports `unremovePatch` for now, so if the request is sent except the `patch`, I thought we need to give back `BAD_REQUEST`.

Reply:
I think there are two cases involved here.

- The JSON document is a JSON patch but is not equal to `unremovePatch`, which would be `CONFLICT`.
- The JSON document is not a JSON patch, which would be `BAD_REQUEST`.

Could we send `BAD_REQUEST` when it's not a JSON patch or `CONFLICT` otherwise? You could try to deserialize the content into `JsonPatch` to test if it's a JSON patch.